Assets/Scripts/EOSSessionsManager.cs

```diff
@@ -1862,5 +2021,419 @@ public void DeclineLobbyInvite()
+                return;
+            }
+
+            if (nextPacketSizeBytes == 0)
+            {
+                return;
+            }
+
+            byte[] data = new byte[nextPacketSizeBytes];
+            var dataSegment = new ArraySegment<byte>(data);
+            ProductUserId peerId = null;
+            SocketId socketId = new SocketId() { SocketName = P2PSessionStatusSocketName };
+
+            Result receivePacketResult = EOSManager.Instance.GetEOSPlatformInterface().GetP2PInterface().ReceivePacket(ref options, ref peerId, ref socketId, out byte outChannel, dataSegment, out uint bytesWritten);
+
+            if (receivePacketResult != Result.Success)
+            {
+                Debug.LogErrorFormat("EOSSessionsManager (HandleReceivedP2PMessages): error while reading received packet data, code: {0}.", nextPacketSizeResult);
+                return;
+            }
+
+            if (!peerId.IsValid())
+            {
+                Debug.LogErrorFormat("EOSSessionsManager (HandleReceivedP2PMessages): ProductUserId peerId is not valid!");
+                return;
+            }
+
+            string message = System.Text.Encoding.UTF8.GetString(data);
+
+            Log($"EOSSessionsManager (HandleReceivedP2PMessages): Received a message: {message}");
+
+            if (!message.StartsWith(P2PInformSessionMessageBase))
+            {
+                Debug.LogError($"EOSSessionsManager (HandleReceivedP2PMessages): This function is handling a received message that it wasn't intended to. Perhaps there's a socket or channel conflict? Message: {message}");
+                return;
+            }
+
+            // Match the message to look for (sessionid) (userid) (message element)
+            Match regexMatchOfMessage = Regex.Match(message, @"\(([^\)]*)\) \(([^\)]*)\) \(([^\)]*)\)");
+
+            if (regexMatchOfMessage.Groups.Count != 4)
+            {
+                Debug.LogError($"EOSSessionsManager (HandleReceivedP2PMessages): Message received, but it didn't contain the expected three parts. Message: {message}");
+                return;
+            }
+
+            string sessionId = regexMatchOfMessage.Groups[1].Value;
+            string userId = regexMatchOfMessage.Groups[2].Value;
+            string messageElement = regexMatchOfMessage.Groups[3].Value;
+
+            Log($"EOSSessionsManager (HandleReceivedP2PMessages): Message parts: [{nameof(sessionId)}: {sessionId}] [{nameof(userId)}: {userId}] [{nameof(messageElement)}: {messageElement}]");
+
+            Session session = GetSessionById(sessionId);
+            ProductUserId messagingUserId = ProductUserId.FromString(userId);
+
+            if (session == null)
+            {
+                Debug.LogError($"EOSSessionsManager (HandleReceivedP2PMessages): Message received regarding sessionId {sessionId}, but no local sessions have that id. Message: {message}");
+                return;
+            }
```

Comment:
This is another good example of careful organization of conditionals that helps reduce indentation and increase code clarity.